holochain-introduction-for-developers/holochain-longer-introduction.md

```diff
@@ -0,0 +1,190 @@
+* managing all the user's running DNA instances and web-based GUI bundles
+* installing new DNA and web-based GUI bundles
+* keeping track of capability tokens and controlling access to zome functions via JSON-RPC or bridges
+* responding to JSON-RPC calls from front ends
+* storing source chains and DHT data
+* communicating with other nodes via gossip
+
+Your front ends can also access some of these features through a separate JSON-RPC interface called the 'admin interface'. They can act on the user's behalf to download and install DNA and UI bundles, fork new instances of existing DNA, and generate new identities.
+
+### Delivering your application to the world
+
+#### As a traditional desktop or mobile app
+
+The most basic way to make your app available to users is to bundle the Holochain runtime, the necessary DNA instances, and an appropriate GUI into an installer, and make it available for download on your website, popular download sites, and official app stores. Holochain runs on Windows, macOS, and Linux, and we're working on Android support too.
+
+#### With Holochain
+
+We're building an app store app and a DNA/GUI delivery app. You can expect these core services to be available on every installed Holochain instance, so you can deploy new and updated DNA and GUI bundles to any participating user.
+
+#### With Holo Host
+
+We recognize that not everyone will be running Holochain, especially at the beginning. So we're building a distributed hosting marketplace called Holo Host. You mark your app as Holo-ready, submit it to the app store, and allow a cloud of machines to host DNA instances and serve UI bundles to end-users over the traditional web. Users are still in control of their identities and source chains, but hosts do the heavy work of data storage, asset delivery, and DHT communication.
+
+#### As a hybrid app
+
+If your app has storage and performance needs that make it inappropriate for users to install Holochain on their machines --- a mobile video sharing app, for instance --- consider deploying your DNAs on Holo Host without a web-based UI bundle. Currently our Holo 'light client' library is written in JavaScript, which would make it appropriate for webview or Cordova-based applications.
+
+## Pithy closing statement or call-to-action
+
+---
+---
+---
+---
+---
+
+# Missing subjects
+
+* Core services and apps -- DPKI, Vault (touched on HCHC briefly under [delivering section](#Delivering-your-application-to-the-world)). I've heard that there are also core 'template apps' (for lack of a better phrase). I don't actually know what these are, but they seem to be things like basic DNAs for messaging, commenting, etc.
+* Details of DPKI and what it's supposed to do -- this will be mostly uninteresting from people coming from the server-based world (except that they won't have to learn how to store passwords, which is cool). But it will be an amazing epiphany for people coming from the blockchain world.
```

Comment:
Actually - the implementation is different than a server app. There is no centralized database or directory - so therefore a unique design pattern exists along with personas, membranes, etc. That needs to be a separate document most likely and therefore a link here and summarizing sentence referencing the link.

Reply:
For sure. This is trying to be the briefest introduction to all the puzzle pieces, so linking is the right thing here. I'm starting to feel like DPKI is a big selling point for both server-based developers (no more data leaks!) and blockchain developers (no lost keys!)